pkg/util/archive/zip.go

```diff
@@ -0,0 +1,258 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package archive provides functions to archive and unarchive files.
+package archive
+
+import (
+	"archive/zip"
+	"compress/flate"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+// Zip is an adapted implementation of (*Zip).Archive from
+// https://github.com/mholt/archiver/blob/v3.5.1/zip.go#L140
+// under MIT License.
+func Zip(sources []string, destination string) error {
+	if !strings.HasSuffix(destination, ".zip") {
+		return fmt.Errorf("%s must have a .zip extension", destination)
+	}
+	if fileExists(destination) {
+		return fmt.Errorf("file already exists: %s", destination)
+	}
+	dir := filepath.Dir(destination)
+	if !fileExists(dir) {
+		err := os.MkdirAll(dir, 0755)
+		if err != nil {
+			return fmt.Errorf("%w making directory: %s", err, dir)
+		}
+	}
+
+	out, err := os.Create(destination)
+	if err != nil {
+		return fmt.Errorf("%w creating %s", err, destination)
+	}
+	defer out.Close()
+
+	zipW := zip.NewWriter(out)
+	zipW.RegisterCompressor(zip.Deflate, func(out io.Writer) (io.WriteCloser, error) {
+		return flate.NewWriter(out, flate.DefaultCompression)
+	})
+	defer zipW.Close()
+
+	for _, source := range sources {
+		err := writeWalk(zipW, source, destination)
+		if err != nil {
+			return fmt.Errorf("%w walking %s", err, source)
+		}
+	}
+
+	return nil
+}
+
+// UnZip unpacks the .zip file at source to destination.
+func UnZip(source, destination string) error {
```

Comment:
Did you write that from scratch ? If not I guess we need to link to the original function (eg. `Unarchive`)

Reply:
I wrote that from scratch